examples/magic_parameter_sweep.py

```diff
@@ -0,0 +1,39 @@
+"""Example showing how to accomplish a napari parameter sweep with magicgui.
+
+It demonstrates:
+1. overriding the default widget type with a custom class
```

Comment:
It would be good to tell users what the advantages of using a custom class here are, and in which circumstances it's useful to do.

Reply:
I might be inclined to change this example then until after the next magicgui release, maybe even hold off on this example entirely, as adding sliders will be a big thing people want to do and I'm not sure we want to put out a more complex way to do it that we know will change to a simpler one

Reply:
I agree.  One of the most common "mis-uses" of magicgui I've seen is people providing `widget_type: QSpinBox` ... even if the parameter is an `int` ... which makes me die a little inside :) cause the whole point was to not have to use QtWidgets directly.  And I think it's probably because of this slider example... so yeah, I'd hate to see that pattern proliferate.

Reply:
This has been changed with the merge of https://github.com/napari/magicgui/pull/43